### ISSUE TITLE
Unobserved Exception thrown while running cache refresh task

### ISSUE THREAD

**Issue by aavasthy** [Azure/azure-cosmos-dotnet-v3#4164]
>We are continuously addressing and improving the SDK, if possible, make sure the problem persist in the [latest SDK version](https://www.nuget.org/packages/Microsoft.Azure.Cosmos).

**Describe the bug**
User recently upgraded their service's CosmosDB SDK from v3.32.3 to v3.33.0 and is experiencing unhandled ArgumentExceptions from the CosmosDB SDK occasionally.

Logs:
```
[OnUnhandledException,UnhandledExceptionReporter.cs(37) <NoCorrelationId> ] An unhandled exception has occurred. The process may be in an inconsistent state. Context: TaskScheduler_UnobservedTaskException CorrelationId:
ExpInfo(fullInfo->System.AggregateException: A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread. (Identifier {0} does not exist. Please call start before calling end. (Parameter '7fe193c0-a931-4f29-a8b2-9af6e849b62a'))
 ---> System.ArgumentException: Identifier {0} does not exist. Please call start before calling end. (Parameter '7fe193c0-a931-4f29-a8b2-9af6e849b62a')
   at Microsoft.Azure.Cosmos.Tracing.TraceData.ClientSideRequestStatisticsTraceDatum.RecordAddressResolutionEnd(String identifier)
   at Microsoft.Azure.Cosmos.Routing.GatewayAddressCache.GetServerAddressesViaGatewayAsync(DocumentServiceRequest request, String collectionRid, IEnumerable`1 partitionKeyRangeIds, Boolean forceRefresh)
   at Microsoft.Azure.Cosmos.Routing.GatewayAddressCache.GetAddressesForRangeIdAsync(DocumentServiceRequest request, PartitionAddressInformation cachedAddresses, String collectionRid, String partitionKeyRangeId, Boolean forceRefresh)
   at Microsoft.Azure.Cosmos.AsyncCacheNonBlocking`2.AsyncLazyWithRefreshTask`1.CreateAndWaitForBackgroundRefreshTaskAsync(Func`2 createRefreshTask)
   at Microsoft.Azure.Cosmos.AsyncCacheNonBlocking`2.UpdateCacheAndGetValueFromBackgroundTaskAsync(TKey key, AsyncLazyWithRefreshTask`1 initialValue, Func`2 callbackDelegate, String operationName)
   at Microsoft.Azure.Cosmos.AsyncCacheNonBlocking`2.RefreshAsync(TKey key, Func`2 singleValueInitFunc)
   at Microsoft.Azure.Cosmos.Routing.GatewayAddressCache.<>c__DisplayClass25_0.<<TryGetAddressesAsync>b__5>d.MoveNext()
   --- End of inner exception stack trace ---) 
```
Needs to investigate:
- Root cause for unobserved task exceptions and can it be avoided.
- Why is this issue prevalent in certain version v3.33 and not in other v3.32/.

**Environment summary**
SDK Version: v3.33
OS Version (e.g. Windows, Linux, MacOSX)

**Additional context**
Add any other context about the problem here (for example, complete stack traces or logs).

**Comment by ealsur** [comment]
@aavasthy 
Isn't this the same as https://github.com/Azure/azure-cosmos-dotnet-v3/issues/4010?

**Comment by ealsur** [comment]
It was released in 3.36.0

**Comment by aavasthy** [comment]
@kundadebdatta the issue looks the same as this https://github.com/Azure/azure-cosmos-dotnet-v3/issues/4010. Please confirm if the fix is in 3.36.0.

**Comment by aavasthy** [comment]
Duplicate issue: https://github.com/Azure/azure-cosmos-dotnet-v3/issues/4010
